### PR TITLE
A bit more comprehensive readme

### PR DESCRIPTION
**Pull request by prateekbh**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -3,30 +3,47 @@
 # AMP Service worker
 AMP service worker is a service worker library for your AMP pages.
 
-The project aims to bring network resiliency and other network related optimizations to any AMP page which is being controlled by this service worker.
-Core use cases:
+**Purpose:** The project aims to bring network resiliency and other network related optimizations to any AMP page which is being controlled by this service worker.
```
Comment (kristoferbaxter) [inline readme.md]: Perhaps...

Purpose: Enhance network resiliency and optimize AMP asset reuse for AMP documents.

File: readme.md [hunk 2]
```diff
@@ -3,30 +3,47 @@
...
+
+**Core use cases:**  
 - Cache AMP scripts with a `stale-while-revalidate` strategy for a longer duration that http-headers.
```
Comment (kristoferbaxter) [inline readme.md]: "for a longer duration than the default http response headers indicate."

File: readme.md [hunk 3]
```diff
@@ -3,30 +3,47 @@
...
 - Cache valid visited AMP documents, and serve only in case of flaky network conditions.
 - Cache assets which are critical to a page with a given strategy.
-- Prefetch outgoing links from an AMP page.
 - Cache an offline page in order to show when a user navigates to a page which was previously not visited.
+- Prefetch outgoing links from an AMP page.
```
Comment (kristoferbaxter) [inline readme.md]: The mixed usage of page and document is slightly confusing. I'd choose one a day stick with it.

File: readme.md [hunk 4]
```diff
@@ -3,30 +3,47 @@
...
 
 
 ## Usage
-In order to use this library user can include the library with an importScript in their service worker
+In order to use AMP-SW you need to add a service worker to your AMP pages. You can follow these steps to do the same and add network resiliency to your AMP pages.
```
Comment (kristoferbaxter) [inline readme.md]: To take advantage of amp-sw, you'll need to host a resource loaded into the browser. Start by following these steps:

File: readme.md [hunk 5]
```diff
@@ -3,30 +3,47 @@
...
 
+- Host a service worker file at the same directory level as your AMP web app/ AMP page.
+- Add [amp-install-serviceworker](https://amp.dev/documentation/components/amp-install-serviceworker) to your AMP pages and point it your service worker file.
```
Comment (kristoferbaxter) [inline readme.md]: For each of these steps and example would be beneficial.

File: readme.md [hunk 6]
```diff
@@ -3,30 +3,47 @@
...
+- Add the following code to your service worker file to get the benefits of AMP-SW out of the box.
 ```
 importScripts('https://cdn.ampproject.org/sw/amp-sw.js');
 AMP_SW.init();
 ```
+- You can add config to `AMP_SW.init()` to customize the behavior of the service worker to fit your needs. e.g. You can cache your static assets, control your document timeout limits or add an offline page. Read more in [modules](#modules) section.
```
Comment (kristoferbaxter) [inline readme.md]: You can configure amp-sw to customize the...

File: readme.md [hunk 7]
```diff
@@ -3,30 +3,47 @@
...
+    url: '/offline.html',
+    assets: [],
+  },
+});
+```
+
 
 ## Modules
 
 The AMP service worker is built up of the following modules:
```
Comment (kristoferbaxter) [inline readme.md]: Calling this a service worker here, but amp-sw everywhere else is unnecessary inconsistency. Possible to choose a name and use it thoughout?

File: readme.md (hunk 6 above)
Comment (prateekbh) [reply readme.md]: done

File: readme.md (hunk 7 above)
Comment (prateekbh) [reply readme.md]: right, I chose the verbose `AMP service worker` for everywhere

File: readme.md (hunk 5 above)
Comment (prateekbh) [reply readme.md]: done

**Pull request merged**